ring/watcher.go

```diff
@@ -0,0 +1,54 @@
+package ring
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Watcher interface {
+	Register() (<-chan interface{}, error)
+	UnRegister()
+	notify()
+}
+
+type watcher struct {
+	mtx      sync.Mutex
+	notifier chan interface{}
```

Comment:
Isn't it more readable if the type of the channel is `struct{}`, i.e. signaling that no information is passed over the channel?
```suggestion
	notifier chan struct{}
```

Reply:
I'm just a noob at golang, fixed